### PR TITLE
Use MkDocs to generate documentation

### LINKED ISSUE #542: Build Quokka documentation with Markdown generated HTML files
**Describe the proposal**
I don't find it beneficial to generate Quokka documentation using Sphinx. Usually, Sphinx is used when you have lots of docstrings/comments in your code and you generate a page from that. However, unless I'm missing something, the [quokka documentation page](https://quokka-astro.github.io/quokka/index.html) doesn't have such content at all. Pretty much everything is hand-written in a way that is much easier to write with Markdown, and I personally believe Markdown is a better way to write stuff. So, why not use Markdown? One of the ways to do this is to use [GitHub Pages](https://pages.github.com/), which is a service that turns Markdown files into a website and hosts them on github. I guess a wiki page inside the Quokka repo is similar, although I've never tried it.

### PR DESCRIPTION
**Pull request by chongchonghe**
### Description

Use MkDocs to generate the documentation page, replacing Sphinx.

I tested it on my computer with the following steps and it worked perfectly fine. Not guaranteed to work on github. May need your help @BenWibking to update the docs.yml properly. 

```sh
cd docs2
python -m pip install -r requirements.txt
mkdocs build && mkdocs serve
```

Currently all the new docs are in docs2. Once everything works fine, I'll use it to replace docs. 

### Related issues
Closes #542 

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [ ] I have added a description (see above).
- [ ] I have added a link to any related issues see (see above).
- [ ] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code.
- [ ] I have tested this PR on my local computer and all tests pass.
- [ ] I have manually triggered the GPU tests with the magic comment `/azp run`.
- [ ] I have requested a reviewer for this PR.

### CODE REVIEW

**Comment by BenWibking** [comment]
It looks like there are some missing images:
```
WARNING -  Doc file 'flowchart.md' contains a link 'flowchart-v2.pdf', but the target is not found among documentation files.
WARNING -  Doc file 'tests/energy_exchange.md' contains a link 'attach/radcoupling_rsla.png', but the target
           'tests/attach/radcoupling_rsla.png' is not found among documentation files.
WARNING -  Doc file 'tests/energy_exchange.md' contains a link 'attach/radcoupling.png', but the target 'tests/attach/radcoupling.png' is not
           found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_temperature-1.png', but the target
           'tests/attach/radhydro_pulse_temperature-1.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_density-1.png', but the target
           'tests/attach/radhydro_pulse_density-1.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_velocity-1.png', but the target
           'tests/attach/radhydro_pulse_velocity-1.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_temperature.png', but the target
           'tests/attach/radhydro_pulse_temperature.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_density.png', but the target
           'tests/attach/radhydro_pulse_density.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_pulse.md' contains a link 'attach/radhydro_pulse_velocity.png', but the target
           'tests/attach/radhydro_pulse_velocity.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_uniform_adv.md' contains a link 'attach/radhydro_uniform_advecting_temperature.png', but the target
           'tests/attach/radhydro_uniform_advecting_temperature.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_uniform_adv.md' contains a link 'attach/radhydro_uniform_advecting_velocity.png', but the target
           'tests/attach/radhydro_uniform_advecting_velocity.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_uniform_adv.md' contains a link 'attach/radhydro_uniform_advecting_temperature-nobeta.png', but the
           target 'tests/attach/radhydro_uniform_advecting_temperature-nobeta.png' is not found among documentation files.
WARNING -  Doc file 'tests/radhydro_uniform_adv.md' contains a link 'attach/radhydro_uniform_advecting_velocity-nobeta.png', but the target
           'tests/attach/radhydro_uniform_advecting_velocity-nobeta.png' is not found among documentation files.
WARNING -  Doc file 'tests/radshock.md' contains a link 'attach/radshock_cgs_temperature.png', but the target
           'tests/attach/radshock_cgs_temperature.png' is not found among documentation files.
WARNING -  Doc file 'tests/shu_osher.md' contains a link 'attach/hydro_shuosher.png', but the target 'tests/attach/hydro_shuosher.png' is not
           found among documentation files.
WARNING -  Doc file 'tests/sms.md' contains a link 'attach/hydro_sms.png', but the target 'tests/attach/hydro_sms.png' is not found among
           documentation files.
```

File: build_docs.sh
Comment (BenWibking) [inline build_docs.sh:1]: ~~Since MkDocs is so easy to use, I think we don't need this file anymore.~~ Actually, nevermind. This is still useful.

**Comment by BenWibking** [comment]
~~The links don't work in the version generated on GitHub...~~

Ok, this is just because I downloaded them and opened the files in the web browser, which doesn't redirect links to the subdirectories to `subdir/index.html`. I think this is not actually an issue.

**Comment by BenWibking** [comment]
@chongchonghe Can you fix the missing images? I'm guessing you just forgot to `git add` them to the PR.

**Comment by chongchonghe** [comment]
@BenWibking  Can you test it now? Should work now.

**Comment by chongchonghe** [comment]
@BenWibking How do you see the html files/pages rendered by github before merging this to the development branch?

**Comment by BenWibking** [comment]
> @BenWibking How do you see the html files/pages rendered by github before merging this to the development branch?

You can download the artifact listed at the bottom of this page: https://github.com/quokka-astro/quokka/actions/runs/10293768978?pr=705

**Comment by BenWibking** [comment]
The "test problems" renders as its own chapter:
Screenshot 2024-08-07 at 9 24 21 PM

Is there a way to get it to render as a drop-down menu like before?

**Comment by BenWibking** [comment]
Otherwise, it looks fine to me. Can you replace the old `docs/` folder with `docs2`?

**Comment by chongchonghe** [comment]
> > @BenWibking How do you see the html files/pages rendered by github before merging this to the development branch?
> 
> You can download the artifact listed at the bottom of this page: https://github.com/quokka-astro/quokka/actions/runs/10293768978?pr=705

Awesome!

**Comment by chongchonghe** [comment]
> The "test problems" renders as its own chapter: Screenshot 2024-08-07 at 9 24 21 PM
> 
> Is there a way to get it to render as a drop-down menu like before?

I don't know a way to do it. The menus are controlled in `mkdocs.yml`, and they are either all expanded or all collapsed by default. Can we leave it as is for now and deal with it later? Currently there aren't many test problems so this is probably fine.  Next, I will try to merge all test problems into one single page, and add a TOC inside that page for quick navigation. That are themes that does that automatically.

File: docs2/docs/Doxyfile
Comment (BenWibking) [inline docs2/docs/Doxyfile:1]: Do we need this file anymore?

File: docs2/docs/Makefile
Comment (BenWibking) [inline docs2/docs/Makefile:1]: Do we need this file?

Comment (chongchonghe) [reply docs2/docs/Makefile:1]: No, we don't.

File: docs2/docs/Doxyfile
Comment (chongchonghe) [reply docs2/docs/Doxyfile:1]: We dont' need this for the current setup, so I'll delete it now. In the future, if we want to use doxygen, we can bring it back. There is [a plugin](https://github.com/pieterdavid/mkdocs-doxygen-plugin) to use doxygen in MkDocs.

File: .gitignore [hunk 1]
```diff
@@ -38,11 +38,17 @@ __pycache__
 # image files
 *.png
 
+/docs/site/*
```
Comment (BenWibking) [inline .gitignore:41]: Should there be a leading `/` here? I think that makes it an absolute path.

File: .gitignore [hunk 2]
```diff
@@ -38,11 +38,17 @@ __pycache__
...
+
 # Documentation build directory
 docs/api/
 docs/doxyoutput/
 _build/
 
+# Include image and pdf files inside docs
+!/docs/docs/flowchart-v2.pdf
+!/docs/docs/tests/attach/*.png
```
Comment (BenWibking) [inline .gitignore:50]: Should these have `/` at the beginning of the path?

**Comment by chongchonghe** [comment]
> ~The links don't work in the version generated on GitHub...~
> 
> Ok, this is just because I downloaded them and opened the files in the web browser, which doesn't redirect links to the subdirectories to `subdir/index.html`. I think this is not actually an issue.

Right. You need a local server to view it with working links. You can do this via `python -m http.server` inside the folder.

**Comment by BenWibking** [comment]
> > ~The links don't work in the version generated on GitHub...~
> > Ok, this is just because I downloaded them and opened the files in the web browser, which doesn't redirect links to the subdirectories to `subdir/index.html`. I think this is not actually an issue.
> 
> Right. You need a local server to view it with working links. You can do this via `python -m http.server` inside the folder.

Ah, yeah, I realized that. Everything looks good to me now.

File: .gitignore (hunk 1 above)
Comment (chongchonghe) [reply .gitignore:41]: `/` makes it directly under the root folder, i.e., only {WORKSPACE}/docs/site will be ignored and {WORKSPACE}/folder1/docs/site won't be ignored.

File: .gitignore (hunk 2 above)
Comment (chongchonghe) [reply .gitignore:50]: Same here.

**Comment by BenWibking** [comment]
~~Except for the `.gitignore` file. I think the leading `/` may make it think those are absolute paths.~~

**Review by BenWibking — APPROVED** [review]

**Pull request merged**